### DISCUSSION TITLE
post_233833

### DISCUSSION THREAD

**Discussion by cincuranet** [cincuranet/blog-comments#1344] [Comments]
imported 

https://www.tabsoverspaces.com/id/233833

**Comment by cincuranet** [comment]
Imported comment written by _Matthew Jones_ on _2020-08-04T17:36:47_:

<p>Very cool! I wonder what this would look like with other collection types, like Stack<t> or Queue<t>?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-08-04T17:53:31_:

<p>Now the question is, are you going to do follow up or do you want me to do it. ;)</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jark_ on _2020-08-04T19:17:30_:

<p>I might have missed it, but what is the code you used to benchmark this?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-08-04T20:26:49_:

<p>The pieces are in the text.</p>

**Comment by cincuranet** [comment]
Imported comment written by _svick_ on _2020-08-04T20:35:57_:

<p>0.0000 ns means that the benchmark wasn't actually doing anything, not even loading the static field. Which is also why it's useful to post full sources of your benchmarks, to identify that kind of problem. :-)</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-08-04T20:50:44_:

<p>Just this and nothing else (and the rest of the benchmarks). Which I believe is correct.</p><p>[Benchmark]<br>public TestArray[] ArrayEmpty()<br>{<br>  return Array.Empty<testarray>();<br>}</p><p>The BDN was showing the "The method duration is indistinguishable from the empty method duration" warning, but that's fine/expected. Because you and I both know that this call will get heavily optimized thanks to the generics and static field.</p><p>Also worth mentioning I got few times 0.0006 or 0.1000 or something like that. But that's still on the measurement error, hence I kept the 0.0000 for niceness (although it might not be technically true).</p>

**Comment by cincuranet** [comment]
Imported comment written by _程凱大_ on _2020-08-04T23:37:27_:

<p>use ToArray on Enumerable.Empty to run Benchmark may not a good idea<br>because Array and IEnumerable they are so difference in inside<br>and Array to IEnumerable still need time to process like ToArray<br>unless return T[] not IEnumerable<t></p>

**Comment by cincuranet** [comment]
Imported comment written by _James Curran_ on _2020-08-08T08:16:53_:

<p>Odd that Array Ctor & CtorInit differ by 10%, when the IL is identical.  I guess that's just the testing noise level.</p><p>I  open up ILSpy to checkout exact what happens in List<>  Ctor and Ctor0 to  make such a big difference.<br>Ctor is just a single reference assignment.<br>Ctor0 has two  if()s (int comparisons), and then the same assignment.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2020-08-08T10:31:11_:

<p>>  "Odd that Array Ctor & CtorInit differ by 10%, when the IL is <br>> identical. I guess that's just the testing noise level. I open up ILSpy</p><p>Yeah that's noise.</p><p>> to checkout exact what happens in List Ctor and Ctor0 to make such a <br>> big difference. Ctor is just a single reference assignment. Ctor0 has <br>> two if()s (int comparisons), and then the same assignment."</p><p>Correct. The code is even linked from the blog post.</p>

**Comment by DreamTreem** [comment]
Great! Thank you.
